keras_nlp/tokenizers/byte_pair_tokenizer.py

```diff
@@ -609,6 +610,13 @@ def detokenize(self, inputs):
             outputs = tf.squeeze(outputs, 0)
         return outputs
 
+    def compute_output_spec(self, input_spec):
+        return keras.Input(
+            input_spec.shape + (self.sequence_length,),
+            dtype=self.compute_dtype,
+            sparse=not self.sequence_length,
```

Comment:
Why are you setting sparse here? It's possible that we need more support in core Keras for this, but in general tokenizers will not return a sparse type in tensorflow, they will return a ragged type. What happens if you just don't set `sparse` here?

Reply:
I believe that was a confusion on my part. I have a (most probably garbled) memory of seeing a KerasNLP tokenizer returning a sparse KerasTensor in the absense of a `sequence_length` and tried to mimic that behavior, but a quick test tells me I was mistaken. Removing the line in question seems to have no effect on the intended functionality.

As a side note, do you know whether keras will support ragged tensors in the forseeable future?

Reply:
I think probably not, but TBD.

Our current read is that while sparse tensors are genuinely useful inside a trainable model code, raggeds are usually useful during preprocessing. For preprocessing we are more interested in moving away from `tf.data` to something more framework agnostic.

But it looks like Torch is adding something [ragged-ish](https://pytorch.org/tutorials/prototype/nestedtensor.html) as a prototype, so who knows? If we have equivalent functionality across backends, and there's some modeling code that would benefit from ragged representation, I would imaging it's a welcome addition. Depends how the whole ml framework ecosystem evolves.